log/log.go

```diff
@@ -0,0 +1,90 @@
+// Package log is an output-library that can print nicely formatted
+// messages to the screen.
+//
+// There are log-level messages that will be printed according to the
+// current debug-level set. Furthermore a set of common messages exist
+// that are printed according to a chosen format.
+//
+// The log-level messages are:
+//	log.Lvl1("Important information")
+//	log.Lvl2("Less important information")
+//	log.Lvl3("Eventually flooding information")
+//	log.Lvl4("Definitively flooding information")
+//	log.Lvl5("I hope you never need this")
+// in your program, then according to the debug-level one or more levels of
+// output will be shown. To set the debug-level, use
+//	log.SetDebugVisible(3)
+// which will show all `Lvl1`, `Lvl2`, and `Lvl3`. If you want to turn
+// on just one output, you can use
+//	log.LLvl2("Less important information")
+// By adding a single 'L' to the method, it *always* gets printed.
+//
+// You can also add a 'f' to the name and use it like fmt.Printf:
+//	log.Lvlf1("Level: %d/%d", now, max)
+//
+// The common messages are:
+//	log.Print("Simple output")
+//	log.Info("For your information")
+//	log.Warn("Only a warning")
+//	log.Error("This is an error, but continues")
+//	log.Panic("Something really went bad - calls panic")
+//	log.Fatal("No way to continue - calls os.Exit")
+//
+// These messages are printed according to the value of 'Format':
+// - Format == FormatLvl - same as log.Lvl
+// - Format == FormatPython - with some nice python-style formatting
+// - Format == FormatNone - just as plain text
+//
+// The log-package also takes into account the following environment-variables:
+//	DEBUG_LVL // will act like SetDebugVisible
+//	DEBUG_TIME // if 'true' it will print the date and time
+//	DEBUG_COLOR // if 'false' it will not use colors
+// But for this the function ParseEnv() or AddFlags() has to be called.
+package log
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"os"
+)
+
+// For testing we can change the output-writer
+var stdOut io.Writer
+var stdErr io.Writer
+
+func init() {
+	stdOut = os.Stdout
+	stdErr = os.Stderr
+}
+
+var bufStdOut bytes.Buffer
+var wrStdOut *bufio.Writer
+var bufStdErr bytes.Buffer
+var wrStdErr *bufio.Writer
+
+func StdToBuf() {
+	wrStdOut = bufio.NewWriter(&bufStdOut)
+	stdOut = wrStdOut
+	wrStdErr = bufio.NewWriter(&bufStdErr)
+	stdErr = wrStdErr
+}
+
+func StdToOs() {
+	stdOut = os.Stdout
+	stdErr = os.Stderr
+}
+
+func GetStdOut() string {
```

Comment:
* `Get` or no `Get` ... I tend for the latter, you ? (i don't really care, just wanna know if it's a decision or just an unintentional act)  
* StdOut is for *standard* output, but here it can also be the buffer output, so ... `OutputString()` & `ErrorString()` ? I don't know, these are not great names but I'm not a fan of `Stdout` for "output stream" and `stderr` for "error stream".

What do you think ?